search/exponential_search.ts

```diff
@@ -1,9 +1,16 @@
+import { binarySearchIterative } from './binary_search'
+
 /**
  * @description Exponential search algorithm for a sorted array.
  *
  * The algorithm searches for a specific value in a sorted array by first finding a range
  * where the value may be present and then performing a binary search within that range.
  *
+ * Compared with binary search, exponential search can be more convenient and advantageous
+ * in cases where the element to be searched is closer to the beginning of the array, 
+ * thus avoiding several comparisons that would make the search more verbose. 
+ * Exponential search doubles the search time with each iteration.
```

Comment:
This last sentence makes no sense. The search time isn't doubled? Do you mean "doubles the (size of the) search *space*"?

Reply:
I was referring to the article featured on geekforgeeks 

The time complexity of exponential search is O(log n) , where n is the size of the array. This means that the search time doubles with each iteration. Binary search , on the other hand, has a time complexity of O(log n), which means that the search time increases logarithmically with the size of the array.

I did not want to be too long-winded. I will delete the last comment if necessary.

Reply:
> The time complexity of exponential search is O(log n)

This is correct. And as you noted, it's the same worst case time complexity as binary search, hence it is important to differentiate the two, otherwise there would be no point in exponential search (since it ends up using binary search anyways).

As you correctly noted, exponential search is advantageous when the element is closer to the beginning. (As an aside: Another use case is when the length of the "array" - the upper bound necessary for binary search - is not known ahead of time and can only be determined on-demand by accessing elements, but that doesn't really apply to this implementation. Then the "exponentiating" step is necessary to find this upper bound to begin with. One example application of this would be finding the first boundary `#t` of a Lua table `t`.)

> This means that the search time doubles with each iteration.

This isn't right. With one more iteration, the execution time is only increased by a small constant factor.

The observation from which the time complexity follows is that if you double the size of the array, exponential search only requires one additional iteration.

> I will delete the last comment if necessary.

It would certainly help to delete it.

Reply:
That makes sense, thank you very much for the explanation!